docs/examples/use_cases/tensorflow/efficientdet/train.py

```diff
@@ -0,0 +1,232 @@
+import multiprocessing
+from absl import logging
+import numpy as np
+import tensorflow as tf
+import random
+import re
+import os
+
+import hparams_config
+import utils
+from model import efficientdet_net
+from model.utils import optimizers
+
+
+def run_training(args):
+    logging.set_verbosity(logging.WARNING)
+
+    args = utils.dict_to_namedtuple(args)
+
+    config = hparams_config.get_efficientdet_config(args.model_name)
+    config.override(args.hparams, allow_new_keys=True)
+    config.image_size = utils.parse_image_size(config.image_size)
+
+    params = dict(
+        config.as_dict(),
+        seed=args.seed,
+        train_batch_size=args.train_batch_size,
+    )
+
+    logging.info(params)
+
+    if args.ckpt_dir:
+        ckpt_dir = args.ckpt_dir
+        if not tf.io.gfile.exists(ckpt_dir):
+            tf.io.gfile.makedirs(ckpt_dir)
+        config_file = os.path.join(ckpt_dir, "config.yaml")
+        if not tf.io.gfile.exists(config_file):
+            tf.io.gfile.GFile(config_file, "w").write(str(config))
+
+    if params["seed"]:
+        seed = params["seed"]
+        os.environ["PYTHONHASHSEED"] = str(seed)
+        tf.random.set_seed(seed)
+        np.random.seed(seed)
```

Comment:
How hard would it be to migrate from the legacy global random state in numpy and go with object-based?
```python
rng = np.random.default_rng(seed)
```

Reply:
Actually, NumPy's randomness isn't used in code explicitly but in case there is NumPy used in imported modules in such a way we need to set seed in the legacy way.